src/actor/npc/ac_npc_shop_master.c

```diff
@@ -0,0 +1,2677 @@
+        if (mEv_CheckEvent(mEv_SAVED_HRATALK_PLR0 + Common_Get(player_no)) == TRUE) {
+            wait_type = aNSC_WAIT_TYPE_HRATALK;
+        } else if (aNSC_check_present_balloon() == TRUE) {
+            wait_type = aNSC_WAIT_TYPE_BALLOON;
+        } else {
+            int idx = mHS_get_arrange_idx(Common_Get(player_no));
+            mHm_rmsz_c *size = &Save_Get(homes)[idx].size_info;
+            if (Now_Private->inventory.loan == 0 && size->renew == FALSE && size->size < 3 && size->size == size->next_size && size->basement_ordered == FALSE) {
+                wait_type = aNSC_WAIT_TYPE_REHOUSE;
+            }
+        }
+    } else if (aNSC_check_present_balloon() == TRUE) {
+        wait_type = aNSC_WAIT_TYPE_BALLOON; 
+    } else {
+        wait_type = aNSC_WAIT_TYPE_3;
+    }
+
+    if (mDemo_Check(mDemo_TYPE_SPEAK, actorx) == TRUE && mDemo_Check_ListenAble() == FALSE) {
+        switch (wait_type) {
+        case aNSC_WAIT_TYPE_HRATALK:
+            mEv_EventOFF(mEv_SAVED_HRATALK_PLR0 + Common_Get(player_no));
+            action = 1;
+            break;
+        case aNSC_WAIT_TYPE_REHOUSE:
+        case aNSC_WAIT_TYPE_BALLOON:
+        case aNSC_WAIT_TYPE_3:
+            action = shop_master->next_action;
+            break;
+        case aNSC_WAIT_TYPE_DONE_REHOUSE:
+            {
+                int idx = mHS_get_arrange_idx(Common_Get(player_no));
+                mHm_hs_c *home = &Save_Get(homes)[idx];
+                switch (home->size_info.size) {
+                case mHm_HOMESIZE_MEDIUM:
+                    if (home->flags.has_basement == FALSE) {
+                        action = 0x6;
+                    } else {
+                        action = 0x4;
+                    }
+                    break;
+                case mHm_HOMESIZE_LARGE:
+                    if (home->flags.has_basement == FALSE) {
+                        aNSC_set_make_basement_info();
+                        action = 0x1;
+                    } else {
+                        action = 0x4;
+                    }
+                    break;
+                default:
+                    action = 0x4;
+                    break;
+                }
+                break;
+            }
+        }
+        aNSC_Set_ListenAble(shop_master);
+        aNSC_setupAction(shop_master, game, action);
+        return;
+    }
+
+
+    proc = (mDemo_REQUEST_PROC)none_proc1;
+    switch (wait_type) {
+    case aNSC_WAIT_TYPE_REHOUSE:
+        proc = aNSC_set_talk_info_start_wait;
+        break;
+    case aNSC_WAIT_TYPE_DONE_REHOUSE:
+        proc = aNSC_set_talk_info_start_wait1;
+        break;
+    case aNSC_WAIT_TYPE_BALLOON:
+        proc = aNSC_set_talk_info_start_wait2;
+        break;
+    case aNSC_WAIT_TYPE_3:
+        proc = aNSC_set_talk_info_start_wait3;
+        break;
+    case aNSC_WAIT_TYPE_HRATALK:
+        proc = aNSC_set_talk_info_start_wait4;
+        break;
+    }
+    mDemo_Request(0x8, actorx, proc);
```

Comment:
These demo functions use the `mDemo_TYPE_*` enum. This one is `mDemo_TYPE_SPEAK`.